### ISSUE TITLE
Doubt about self.output in function create_dc_actor_critic PPO

### ISSUE THREAD

**Issue by icaro56** [Unity-Technologies/ml-agents#958]
Could anyone explain to me why the tf.multinomial function is used with the policy instead of using with the probabilities?

It is like this
```
self.policy = tf.layers.dense(hidden, self.a_size, activation=None, use_bias=False,
                                      kernel_initializer=c_layers.variance_scaling_initializer(factor=0.01))

self.all_probs = tf.nn.softmax(self.policy, name="action_probs")
output = tf.multinomial(self.policy, 1)
self.output = tf.identity(output, name="action")
```
It could be like this.
```
self.policy = tf.layers.dense(hidden, self.a_size, activation=None, use_bias=False,
                                      kernel_initializer=c_layers.variance_scaling_initializer(factor=0.01))

self.all_probs = tf.nn.softmax(self.policy, name="action_probs")
output = tf.multinomial(self.all_probs, 1)
self.output = tf.identity(output, name="action")
```

**Comment by awjuliani** [comment]
HI @icaro56,

The `tf.multinomial` function takes the *un-normalized* log-probabilities, rather than the normalized probabilities, which is what `self.all_probs` is. You can read more about the function here: https://www.tensorflow.org/api_docs/python/tf/multinomial
